src/imperator/tables/on_action.rs

```diff
@@ -0,0 +1,246 @@
+use fnv::FnvHashMap;
+use once_cell::sync::Lazy; // replace with std version once it's stable
+
+use crate::block::BV;
+use crate::context::ScopeContext;
+use crate::everything::Everything;
+use crate::parse::pdxfile::parse_pdx_internal;
+use crate::scopes::{scope_from_snake_case, Scopes};
+use crate::token::Token;
+
+#[derive(Debug, Clone)]
+struct OnActionScopeContext {
+    root: Scopes,
+    names: Vec<(String, Scopes)>,
+    lists: Vec<(String, Scopes)>,
+}
+
+pub fn on_action_scopecontext(key: &Token, _data: &Everything) -> Option<ScopeContext> {
+    if let Some(oa_sc) = ON_ACTION_SCOPES.get(key.as_str()) {
+        let mut sc = ScopeContext::new(oa_sc.root, key);
+        for (name, s) in &oa_sc.names {
+            sc.define_name(name, *s, key);
+        }
+        for (list, s) in &oa_sc.lists {
+            sc.define_list(list, *s, key);
+        }
+        sc.set_strict_scopes(false);
```

Comment:
Lol this line is a bug, and it's a bug in the other two games as well. Should be deleted; this `sc` can be strict because the defined scope names are known.